llama-index-core/llama_index/core/schema.py

```diff
@@ -1037,6 +1094,10 @@ def from_cloud_document(
         )
 
 
+@deprecated(
+    version="0.12.2",
+    reason="'ImageDocument' is deprecated, use 'Document' instead.",
```

Comment:
This might need some more explanation. At this point, users think `Document` is only for text. Maybe we need a docs page to point to?

Reply:
Good point, let's not deprecate it in this PR then, we'll do it once documentation is done.